### DISCUSSION TITLE
Should package references be Visual Studio version specific?

### DISCUSSION THREAD

**Discussion by shruti0085** [microsoft/vs-threading#1002] [Q&A]
Hello,
We have a Visual Studio extension which supports Visual Studio 2017, 2019 and 2022.
We used this doc https://docs.microsoft.com/en-us/visualstudio/extensibility/migration/update-visual-studio-extension?view=vs-2022#modernize-your-vsix-project as the reference to add support for VS 2022 which outlines the concept of shared projects for sharing code across multiple VS Versions. Each vsix project uses version specific VSSDK package references i.e. 16.x and 17.x respectively.

The `Microsoft.VisualStudio.Threading` package versions listed on Nuget also seem to follow the same versioning strategy as the VSSDK packages i.e. 16.x, 17.x. For use with our extension, does this package need to be referenced with a similar version split across respective vsix projects i.e. vsix project targeting 2019 would have a package reference for 16.x and the project targeting 2022 references 17.x?

**Comment by AArnott** [comment]
Yes. If you just reference the Microsoft.VisualStudio.Sdk meta-package, you should get the matching version of vs-threading referenced automatically. But if you use an explicit package reference, its version should not exceed the version of VS that you're targeting. Where a major.minor version of the package appears with several x.y.PATCH versions, you can choose the latest patch and it will be compatible with the x.y version of VS without regard to the patch version.
